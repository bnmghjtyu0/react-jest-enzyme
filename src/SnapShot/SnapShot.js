import React, { Component } from 'react'
import PropTypes from 'prop-types'

class SnapShot extends Component {
    constructor(props) {
        super(props)
    }
    render() {
        return (
            <div>
                <h1>title</h1>
                <p>123</p>
            </div>
        )
    }
}

export default SnapShot