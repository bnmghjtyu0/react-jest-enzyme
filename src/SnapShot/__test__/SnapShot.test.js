import React from 'react'
import { shallow, mount } from "enzyme";
import { findByTestAttr, checkProps } from '../../../test/testUtils';
import sinon from 'sinon';
import SnapShot from '../SnapShot'

const defaultProps = {}

const setup = (props = {}) => {
    const setupProps = { ...defaultProps, ...props }
    return shallow(<SnapShot {...setupProps} />)
}

describe('props function', () => {
    test('render a label', () => {
        const wrapper = setup()
        expect(wrapper).toMatchSnapshot();
    })
})
