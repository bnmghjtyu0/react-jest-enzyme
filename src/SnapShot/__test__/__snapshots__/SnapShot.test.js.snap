// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props function render a label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SnapShot />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          title
        </h1>,
        <p>
          123
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
        },
        "ref": null,
        "rendered": "title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "123",
        },
        "ref": null,
        "rendered": "123",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            title
          </h1>,
          <p>
            123
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "123",
          },
          "ref": null,
          "rendered": "123",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
